src/aae_exchange.erl

```diff
@@ -899,39 +955,33 @@ intersect_ids(IDs0, IDs1) ->
 %% Select a cluster of IDs if the list of IDs is smaller than the maximum 
 %% output size.  The lookup based on these IDs will be segment based, so it 
 %% is expected that the tightest clustering will yield the most efficient 
-%% results. 
-select_ids(IDList, MaxOutput, StateName, ExchangeID, LogLevels) ->
+%% results.  However, if we always get the same list, then concurrent exchanges
+%% will wastefully correct the same data - so randomly chose one of the better
+%% lists
+select_ids(IDList, MaxOutput, StateName, ExchangeID, LogLevels)
+                                            when length(IDList) > MaxOutput ->
     IDList0 = lists:usort(IDList),
+    aae_util:log("EX005", 
+                    [ExchangeID, length(IDList0), StateName],
+                    logs(),
+                    LogLevels),
+    IDList1 =
```

Comment:
`list:nthtail` is probably not available in R16, but the kind of thing you want here to improve readability.
Otherwise `lists:split` is a good alternative
Something similar to:
```
{MyIDList1, IDList2} = list:split(MaxOutput, IDList0) 
```
but IDList1 in the code is all but MyIDList1

Reply:
As discussed, code is currently as expected - the strange split is necessary to do the zip and calcuclate the distance for the `nth` item between itself and the `nth + MaxOutput` item.